shell/browser/api/electron_api_web_frame_main.cc

```diff
@@ -37,31 +37,33 @@
 
 namespace {
 
+using LifecycleState = content::RenderFrameHostImpl::LifecycleStateImpl;
+
+// RenderFrameCreated is called for speculative frames which may not be
+// used in certain cross-origin navigations. Invoking
+// RenderFrameHost::GetLifecycleState currently crashes when called for
+// speculative frames so we need to filter it out for now. Check
+// https://crbug.com/1183639 for details on when this can be removed.
+[[nodiscard]] LifecycleState GetLifecycleState(content::RenderFrameHost* rfh) {
+  auto* rfh_impl = static_cast<content::RenderFrameHostImpl*>(rfh);
```

Comment:
_nit_ `rfh` doesn't have to be mutable here
```suggestion
[[nodiscard]] LifecycleState GetLifecycleState(const content::RenderFrameHost* rfh) {
  const auto* rfh_impl = static_cast<const content::RenderFrameHostImpl*>(rfh);
```